### PR TITLE
Feat/theme change

### PR DESCRIPTION
**Pull request by ArthurEncr**
# Description
Implemented changing themes, between light mode and dark mode. I had to change several project files to use token variables. Theme switch button has been added to the "my profile" menu temporarily.

- Theme changing logic created 
- Button to change theme at my profile menu
- I changed some lines that used colors without being project variables
- New token created (background-left-panel)
- New variable stored in local storage. (mode)
- New prop added to Panel Section name.

[
Jira Ticket
](https://apps.nrs.gov.bc.ca/int/jira/browse/FSADT2-411?)
## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Manually tested and unit tests.

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

I added the change theme button in the "my profile" menu, but when this is developed in the mockup we will change places. I had to create the new token for the left panel after discussions with the UX team. Another task needs to be created to apply the dark theme to notification components, tags and buttons.

### CODE REVIEW

**Review by RMCampos — CHANGES_REQUESTED** [review]
Good job! It looks amazing. However when logging out with dark theme activated the landing page also goes dark. It should keep white.

**Review by lucasbuenoencora — COMMENTED** [review]
Great Job! Just check the unit tests of BCHeader and App that are failing in the unit tests check.

File: src/views/Dashboard/dashboard.tsx [hunk 1]
```diff
@@ -4,6 +4,7 @@ import {
   FlexGrid,
   Row,
   Stack
+
```
Comment (mgaseta) [inline src/views/Dashboard/dashboard.tsx:7]: Remove empty line.

File: src/views/Dashboard/dashboard.tsx [hunk 2]
```diff
@@ -23,6 +24,7 @@ const Dashboard = () => (
       <RecentActivities />
     </Stack>
   </FlexGrid>
+
```
Comment (mgaseta) [inline src/views/Dashboard/dashboard.tsx:27]: Remove empty line.

**Review by mgaseta — CHANGES_REQUESTED** [review]
Good job! Just 2 small markups (Besides Lucas' notes). 🚀

File: src/utils/ThemePreference.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+interface ThemeContextData {
+  theme: string,
+  setTheme: React.Dispatch<React.SetStateAction<string>>
+}
+
+const ThemePreferenceContext = createContext<ThemeContextData>({} as ThemeContextData);
+const currentMode = localStorage.getItem('mode');
+
+/** */
```
Comment (mgaseta) [inline src/utils/ThemePreference.tsx:14]: Fix documentation here.

File: src/utils/ThemePreference.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+interface ThemePreferenceProps {
+  children?: ReactNode
+}
+
+/**
+ * @param root0
+ * @param root0.children
+ */
```
Comment (mgaseta) [inline src/utils/ThemePreference.tsx:26]: Fix documentation here.

**Review by mgaseta — CHANGES_REQUESTED** [review]
Good job! Just some small changes regarding code documentation. Also, take a look at the conflicts.

File: src/utils/ThemePreference.tsx [hunk 5]
```diff
@@ -1,24 +1,36 @@
...
-const ThemePreferenceContext = createContext();
+interface ThemeContextData {
+  theme: string,
+  setTheme: React.Dispatch<React.SetStateAction<string>>
+}
+
+const ThemePreferenceContext = createContext<ThemeContextData>({} as ThemeContextData);
 const currentMode = localStorage.getItem('mode');
 
+/** */
```
Comment (RMCampos) [inline src/utils/ThemePreference.tsx:14]: I think you forgot to add the function description here.

File: src/utils/ThemePreference.tsx [hunk 6]
```diff
@@ -1,24 +1,36 @@
...
 }
 
-function ThemePreference({ children }) {
+interface ThemePreferenceProps {
+  children?: ReactNode
+}
+
+/**
+ * @param root0
+ * @param root0.children
```
Comment (RMCampos) [inline src/utils/ThemePreference.tsx:25]: You could use references to the parameters, using a notation like `children {ReactNode} nodes to be rendered.` Also don't forget to include a description to the function.

**Review by RMCampos — CHANGES_REQUESTED** [review]

**Comment by RMCampos** [comment]
Good job! 🥳 Only few things related to JSDoc.

**Review by mgaseta — APPROVED** [review]
LGTM!

**Review by lucasbuenoencora — APPROVED** [review]
Great Job!

**Review by RMCampos — APPROVED** [review]
Looks perfect! 🥇

**Pull request merged**